tests/child-wait.test.sh

```
@@ -560,12 +620,16 @@ _uberdev_dispatch_cancel_backend background "$PID_GUARD" "$IDENTITY"
 # non-zombie members remain. A mismatched launch identity never signals either
 # the wrapper or its resistant child.
 python3 -I -c 'import os; os.setsid(); os.execvp("bash",["bash","-c","trap \"\" TERM; (trap \"\" TERM; while :; do :; done) & wait"])' & TERM_GUARD=$!
+CHILD_WAIT_GUARD_GROUPS="$CHILD_WAIT_GUARD_GROUPS $TERM_GUARD"
 disown "$TERM_GUARD" 2>/dev/null || true
-for _ in 1 2 3 4 5 6 7 8 9 10; do TERM_CHILD="$(pgrep -P "$TERM_GUARD" | head -1 || true)"; [ -z "$TERM_CHILD" ] || break; done
-[ -n "$TERM_CHILD" ] && kill -0 "$TERM_CHILD"
+TERM_CHILD="$(_child_wait_await_provider_child "$TERM_GUARD" 'TERM-resistant cancel')"
+kill -0 "$TERM_CHILD"
 TERM_IDENTITY="$(_uberdev_agent_process_identity "$TERM_GUARD")"
 ! _uberdev_dispatch_cancel_backend background "$TERM_GUARD" "${TERM_IDENTITY%|*}|reused"
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)